Automattic/sensei
# Incorrect lesson layout sometimes added in editor

## Proposed Changes
This PR fixes an issue where clicking the _Start with default layout_ link in the _Lesson Layout_ modal would add the wrong template if the link and the template beneath it overlapped (see screenshot). The solution was to add a `z-index` to the modal's footer so that the link is on top and will get the click.

![Screenshot 2023-09-11 at 11 29 04 AM](https://github.com/Automattic/sensei/assets/1190420/ab2697f4-9f1f-4f35-843d-a51e1c1c62cc)

## Testing Instructions


1. Create a new lesson.
2. In the _Lesson Layout_ modal, zoom in until the template overlaps with the _Start with default layout_ link.
3. Click the _Start with default layout_ link.
4. Ensure the default layout template is added in the editor.

## Pre-Merge Checklist
